Codecademy/gamut
# fix(Button): button typing issues

### Overview


Correct Button types to not be overly permissive


### PR Checklist

- [x] Related to designs: no
- [x] Related to JIRA ticket: [GM-TBA]

#### Testing Instructions

1. Check that buttons still look the same in Storybook preview
2. Check out this branch locally
3. Uncomment the test code and see that the correct errors are reported by TS
4. Profit ✨ 

#### PR Links and Envs

* [Monorepo PR](https://github.com/codecademy-engineering/mono/pull/4445) (in progress)
* Monolith TBD

I will not be shipping this until the above are also reviewed + ready to ship
